ui/apps/platform/src/Containers/Collections/CollectionsFormPage.tsx

```diff
@@ -1,39 +1,64 @@
-import React, { ReactElement, useCallback } from 'react';
-import { PageSection } from '@patternfly/react-core';
+import React, { ReactElement, useCallback, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import {
+    Alert,
+    AlertActionCloseButton,
+    AlertGroup,
+    Breadcrumb,
+    BreadcrumbItem,
+    Button,
+    Divider,
+    Dropdown,
+    DropdownItem,
+    DropdownSeparator,
+    DropdownToggle,
+    Flex,
+    FlexItem,
+    Title,
+} from '@patternfly/react-core';
 import { useMediaQuery } from 'react-responsive';
 
 import useRestQuery from 'Containers/Dashboard/hooks/useRestQuery';
 import {
     CollectionResponse,
+    createCollection,
+    deleteCollection,
     getCollection,
     listCollections,
     ResolvedCollectionResponse,
+    updateCollection,
 } from 'services/CollectionsService';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
+import { collectionsBasePath } from 'routePaths';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import ConfirmationModal from 'Components/PatternFly/ConfirmationModal';
+import useToasts from 'hooks/patternfly/useToasts';
+import { values } from 'lodash';
 import { CollectionPageAction } from './collections.utils';
 import CollectionForm from './CollectionForm';
-import { parseCollection } from './converter';
+import { generateRequest } from './converter';
 import { Collection } from './types';
 
 export type CollectionsFormPageProps = {
     hasWriteAccessForCollections: boolean;
     pageAction: CollectionPageAction;
 };
 
-const noopRequest = {
-    request: Promise.resolve(undefined),
-    cancel: () => {},
-};
-
-const defaultCollectionData: Collection = {
+const defaultCollectionData: Omit<CollectionResponse, 'id'> = {
```

Comment:
You keep teaching me more TypeScript. What are pro and con versus `id: ''` property?

Reply:
The main benefit is that this is explicitly saying we don't care that the `CollectionResponse` object is missing the `id` property (and in fact this forbids it from happening). Setting an empty value of `id: ''` would allow mixing of objects in the code that either have a valid id or have an empty string. To prevent bugs when we wanted to access the id we would need to add `if (id === '') {}` runtime checks, while using `Omit` enforces this at compile time.